gcc/rust/backend/rust-compile-expr.cc

```diff
@@ -2029,6 +2029,17 @@ CompileExpr::array_copied_expr (location_t expr_locus,
   unsigned HOST_WIDE_INT len
     = wi::ext (max - min + 1, precision, sign).to_uhwi ();
 
+  unsigned int res;
+  if (__builtin_umul_overflow (TREE_INT_CST_ELT (TYPE_SIZE_UNIT (array_type),
```

Comment:
looks like there is a regression here on running on 32 bit platforms. So this __builtin wont work.

Its definetlyy worth checking this in C/C++ on compiler explorer and if they make the same error and work back from that error message.